docs/_ext/link_roles.py

```diff
@@ -0,0 +1,56 @@
+# based on http://protips.readthedocs.io/link-roles.html
+
+#from __future__ import print_function, unicode_literals
+
+import os
+import re
+import subprocess
+from collections import namedtuple
+
+from docutils import nodes
+from sphinx.transforms.post_transforms import SphinxPostTransform
+
+
+class translation_link(nodes.Element):
+    """Node for "link_to_translation" role."""
+
+
+# Linking to translation is done at the "writing" stage to avoid issues with the info being cached between builders
+def link_to_translation(name, rawtext, text, lineno, inliner, options={}, content=[]):
+    node = translation_link()
+    node['expr'] = (rawtext, text, options)
+    return [node], []
+
+
+class TranslationLinkNodeTransform(SphinxPostTransform):
+    # Transform needs to happen early to ensure the new reference node is also transformed
+    default_priority = 0
+
+    def run(self, **kwargs):
+
+        # Only output relative links if building HTML
+        for node in self.document.traverse(translation_link):
+            if 'html' in self.app.builder.name:
+                rawtext, text, options = node['expr']
+                (language, link_text) = text.split(':')
+                env = self.document.settings.env
+                docname = env.docname
+                doc_path = env.doc2path(docname, False)
+                urlpath = os.environ['LVGL_URLPATH']+'/'
+                return_path = "https://lvgl.100ask.net/"+urlpath
```

Comment:
Can you please move the hard-coded URL to top of this Python file as a variable (constant)?
So the URL will not be "hidden" in source code.

Like this:
```
URL_BASE = 'https://lvgl.100ask.net/'
....
    url_path = os.environ['LVGL_URLPATH'] + '/'
    return_path = URL_BASE + url_path

```